File: Royaleagle73/A1_Q1.clj
(def input [1 2 3 4 5])
(def N 2)

(defn replaceCube [input] 
  (for [value input]
    (if (= (- (/ value N) (int (/ value N))) 0)
      (* value (* value value))
      (* value 1))))

(replaceCube input)

File: Royaleagle73/A1_Q1.c
#include <stdio.h>
void replace_cube(int a[],int n)
{
  int i,count=1;
  for(i=0;i<5;i++)
  {
    if(i+1==n)
    {
      a[i]=(a[i]*a[i]*a[i]);
      count=count+1;
      n=n*count;
    }
  }
}
int main()
{
    int a[5]={1,2,3,4,5},n=2,i;
    replace_cube(a,n);
    for(i=0;i<5;i++)
    {
      printf("%d\t",a[i]);
    }
    return 0;
}
